### PR TITLE
Sanitize path traversal characters from file names before unarchiving them

### PR DESCRIPTION
**Pull request by EthanArbuckle**
This PR aims to eliminate the [Zipperdown](https://zipperdown.org/) security vulnerability. [Relevant Github issue](https://github.com/ZipArchive/ZipArchive/issues/453)

Files inside of an archive may contain path traversal characters in their filename. When unarchiving these, SSZipArchive follows `../` sequences which can lead to a file being written to an unexpected destination. This could potentially lead to RCE under the worst of circumstances (such as overwriting a javascript file that the app is going to execute).

This addition sanitizes dot-dot-slash (`../`) characters from the destination path of files being unarchived, mimicking the default behavior of the Archive Utility found on macOS (as well as most other popular desktop archiving clients). 

Consider an archive containing the following file:
	`../../../../../../../../../../../Users/test/Desktop/hello.txt`
	
When unarchived using the current SSZipArchive, the directory traversal characters are followed and extracted files break out of the current directory, resulting in `/Users/test/Desktop/hello.txt` being created.

Compare this to the default behavior of macOS's Archive Utility, which does _not_ follow directory traversal characters. The same archive results in extraction restricted to the current directory:
`%current_path%/Users/test/Desktop/hello.txt`
	
	
From a security standpoint, discarding directory traversal characters should be the default behavior of this SDK.

### CODE REVIEW

**Comment by jhudsonWA** [comment]
The team will discuss. Thanks for this. @Coeur Thoughts?

**Comment by EthanArbuckle** [comment]
Hey guys, any updates on getting this fix merged in? @jhudsonWA

**Comment by jhudsonWA** [comment]
Sorry all, I will go ahead and pull this in today and make a new release.

**Comment by EthanArbuckle** [comment]
Awesome, thanks! I'll lookout for the version bump and release.

File: SSZipArchive/SSZipArchive.m [hunk 1]
```diff
@@ -393,6 +393,12 @@ + (BOOL)unzipFileAtPath:(NSString *)path
                 strPath = [strPath stringByReplacingOccurrencesOfString:@"\\" withString:@"/"];
             }
             
+            // Sanitize path traversal characters if they're present in the file name to prevent directory backtracking. Ignoring these characters mimicks the default behavior of the Unarchiving tool on macOS.
+            if ([strPath rangeOfString:@"../"].location != NSNotFound) {
```
Comment (Coeur) [inline SSZipArchive/SSZipArchive.m]: @EthanArbuckle , @jhudsonWA , If filename is exactly "." or "..", don't we risk to rewrite the permissions/date for destination folder or its parent?

File: SSZipArchive/SSZipArchive.m [hunk 2]
```diff
@@ -393,6 +393,12 @@ + (BOOL)unzipFileAtPath:(NSString *)path
...
+                // "../../../../../../../../../../../tmp/test.txt" -> "tmp/test.txt"
+                strPath = [[[NSURL URLWithString:strPath] standardizedURL] absoluteString];
```
Comment (Coeur) [inline SSZipArchive/SSZipArchive.m]: Are we supposed to percent-escape some characters before using `URLWithString:`?

**Pull request merged**